front/components/items/Appointment.tsx

```diff
@@ -2,26 +2,38 @@ import WrapperWidth from 'components/wrappers/Wrapperwidth';
 import { isSameDay } from 'date-fns';
 import { strapiAppointmentGet } from 'lib/strapi/appointments/get';
 import { strapiAppointmentQuery } from 'lib/strapi/appointments/queryType';
+import { strapiTherapistGet } from 'lib/strapi/therapists/get';
+import { strapiTherapistQuery } from 'lib/strapi/therapists/queryType';
 import React, { useEffect, useState } from 'react';
 import { DayPicker } from 'react-day-picker';
 import { Grid } from 'react-loader-spinner';
+import Button from './Button';
 
 const Appointment = () => {
-    const TEMPtherapist = 'Bożena';
+    const [chosenTherapist, setChosenTherapist] = useState<string>();
     let rightTime: Date;
     let queryChosenDate: string | undefined;
     const filters: filters = {};
-    filters.therapist = {
-        first_name: {
-            $eq: TEMPtherapist,
-        },
-    };
     let [chosenDate, setChosenDate] = useState<Date>();
     const [isLoading, setIsLoading] = useState(true);
+    const [therapists, setTherapists] = useState<strapiTherapistQuery[]>();
     const [appointmentsTherapist, setAppointmentsTherapist] =
         useState<strapiAppointmentQuery[]>();
     const [appointmentsDate, setAppointmentsDate] =
         useState<strapiAppointmentQuery[]>();
+    useEffect(() => {
+        strapiTherapistGet().then((res) => {
+            setTherapists(res.data.data);
+        });
+    }, []);
+    useEffect(() => {
+        setChosenDate(undefined);
```

Comment:
this useEffect triggers next useEffect so consider removing lines 31-35 which are contained in next useEffect

Reply:
Moreover I'd rather go to places where you use setChosenTherapist and there I'd change chosenData but not sure if its optimal way